### PR TITLE
Fee behavior commit 1

### PR DESCRIPTION
**Pull request by SuheilMakrani**
Please inform me of any changes that i need to make.

### CODE REVIEW

File: features/transaction/announce-a-transaction.feature [hunk 1]
```diff
@@ -71,3 +71,51 @@ Feature: Announce a transaction
 
   # Status errors not treated:
   # - Failure_Core_Too_Many_Transactions
+
+#Fee Behaviour
+  Scenario: An account announced a valid transaction
+    Given Alice announced a valid transaction with "transaction_size" of 10 "nem:xem" to a "MAIN_NET" node with a "fee_multiplier" of 2
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: Please, remove double commas in "transaction_size" and the underscore.

Comment (dgarcia360) [reply features/transaction/announce-a-transaction.feature]: A transaction has a size in bytes, not in nem:xem.

Comment (dgarcia360) [reply features/transaction/announce-a-transaction.feature]: One thing is missing: a transaction should specify the maximum fee willing to pay to be included.

File: features/transaction/announce-a-transaction.feature [hunk 2]
```diff
@@ -71,3 +71,51 @@ Feature: Announce a transaction
...
+    And "effective_fee" is the product of "transaction_size" and "fee_multiplier"
+    When the transaction status is Success
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: When the node process the transaction to include it in a block
Then the node accepts the transaction // Note: transaction::size * block::fee_multiplier should be less or equal than max_fee
And her "xem" balance is deducted in <number> units

File: features/transaction/announce-a-transaction.feature [hunk 3]
```diff
@@ -71,3 +71,51 @@ Feature: Announce a transaction
...
+    Then her "nem:xem" balance is the deducted by the sum of "effective_fee" and the "transaction_size" 
+
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: Add to the Background (line 8): And the native currency mosaic is "xem"

File: features/transaction/announce-a-transaction.feature [hunk 4]
```diff
@@ -71,3 +71,51 @@ Feature: Announce a transaction
...
+    And "effective_fee" is the product of "transaction_size" and "fee_multiplier"
+    And the deadline expired in its unconfirmed state
+    When the transaction status is Success
+    And "effective_fee" is smaller or equal to the "max_fee"
+    Then her "nem:xem" balance is the deducted by the "effective_fee"
+
+  Scenario: An account announced a transaction with an expired deadline
+    Given Alice announced 3 hours ago a transaction with "transaction_size" of 10 "nem:xem" to a "MAIN_NET" node with a "fee_multiplier" of 1
+    And with a deadline of 2 hours
+    And sets a "max_fee" of 25 "nem:xem"
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: Please, replace "nem:xem" appearances for "xem"

**Comment by dgarcia360** [comment]
Thanks @SuheilMakrani for the pull request!

I did a first review of the first scenario, please address the comments to get the PR accepted. Afterward, we will go for a second review.

Please, remember to run ``npm run gherkin-lint`` and fix the issues to check that the  Gherkin code is well formatted.

**Comment by SuheilMakrani** [comment]
Thanks for the response @dgarcia360  I have addressed the comments and made the necessary changes.

File: features/transaction/announce-a-transaction.feature [hunk 5]
```diff
@@ -71,3 +72,42 @@ Feature: Announce a transaction
 
   # Status errors not treated:
   # - Failure_Core_Too_Many_Transactions
+
+  #Fee Behaviour
+  Scenario: An account announced a valid transaction
+    Given Alice announced a valid transaction of 10 "xem" to a "MAIN_NET" node with a "fee_multiplier" of 2
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: The scenario is better, however, I don't know what do you mean by a transaction of 10 "xem". I think you are referring to bytes.

Fixed:

```
   Given Alice announced a valid transaction of size 10 bytes willing to pay 25 xem to include it in a block
    When a node with a fee multiplier of 2 processes the transaction
    Then the node accepts the transaction
    A her "xem" balance is deducted in 20 units.
```

File: features/transaction/announce-a-transaction.feature [hunk 6]
```diff
@@ -71,3 +72,42 @@ Feature: Announce a transaction
 
   # Status errors not treated:
   # - Failure_Core_Too_Many_Transactions
+
+  #Fee Behaviour
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: Please, remove this comment.

File: features/transaction/announce-a-transaction.feature [hunk 7]
```diff
@@ -71,3 +72,42 @@ Feature: Announce a transaction
...
+  Scenario: An account announced a valid transaction
+    Given Alice announced a valid transaction of 10 "xem" to a "MAIN_NET" node with a "fee_multiplier" of 2
+    And sets a "max_fee" of 25 "xem"
+    When the node process the transaction to include it in a block
+    Then the node accepts the transaction // Note: transaction::size * block::"fee_multiplier" should be less or equal than "max_fee"
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: Please, remove the "// Note: transaction::size * block::"fee_multiplier" should be less or equal than "max_fee" (applies to all the document)

File: features/transaction/announce-a-transaction.feature [hunk 8]
```diff
@@ -71,3 +72,42 @@ Feature: Announce a transaction
...
+    And her "xem" balance is deducted in units
+
+  Scenario: An account announced a valid transaction and sets a max fee less than than effective fee
+    Given Alice announced a valid transaction of 10 "xem" to a "MAIN_NET" node with a "fee_multiplier" of 2
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: Replace with suggested modification (first comment)

File: features/transaction/announce-a-transaction.feature [hunk 9]
```diff
@@ -71,3 +72,42 @@ Feature: Announce a transaction
...
+    And sets a "max_fee" of 2 "xem"
+    When transaction::size * block::"fee_multiplier" is greater than "max_fee"
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: When the node with a fee multiplier of 2 processes the transaction
    Then the node rejects the transaction

File: features/transaction/announce-a-transaction.feature [hunk 10]
```diff
@@ -71,3 +72,42 @@ Feature: Announce a transaction
...
+    Then the transaction status will remain unannounced
+    And her "xem" balance stays intact
+
+  Scenario: An expired deadline on unconfirmed transaction
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: Remove this scenario (explanation in next comment).

File: features/transaction/announce-a-transaction.feature [hunk 11]
```diff
@@ -71,3 +72,42 @@ Feature: Announce a transaction
...
+    Given Alice announced a valid transaction of 10 "xem" to a "MAIN_NET" node with a "fee_multiplier" of 1
+    And sets a "max_fee" of 25 "xem"
+    And the deadline expired in its unconfirmed state
+    When the node process the transaction to include it in a block
+    Then the node accepts the transaction // Note: transaction::size * block::"fee_multiplier" should be less or equal than "max_fee"
+    And her "xem" balance is deducted in units
+
+  Scenario: An account announced a transaction with an expired deadline
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: What you should test is what happens when 
1) you send a transaction with max_fee too low,
2) no node that wants to accept the transaction

so the deadline is reached, your transaction rejected and your xem balance is not deducted

File: features/transaction/announce-a-transaction.feature [hunk 12]
```diff
@@ -71,3 +72,42 @@ Feature: Announce a transaction
...
+    Given Alice announced 3 hours ago a transaction of 10 "xem" to a "MAIN_NET" node with a "fee_multiplier" of 1
+    And with a deadline of 2 hours
+    And sets a "max_fee" of 25 "xem"
+    And transaction::size * block::"fee_multiplier" should be less or equal than "max_fee"
+    When the transaction status is Failed due to error "Failure_Core_Past_Deadline"
+    Then her "xem" balance stays intact
+
+  Scenario: An account announced a transaction with an invalid signature
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: The scenario is not needed, An account tries to announce a transaction with an invalid signature is already described in line 40.

File: features/transaction/announce-a-transaction.feature [hunk 13]
```diff
@@ -71,3 +72,42 @@ Feature: Announce a transaction
 
   # Status errors not treated:
```
Comment (dgarcia360) [inline features/transaction/announce-a-transaction.feature]: If you could, move this comment to the end of the file. Thanks!

**Comment by SuheilMakrani** [comment]
Thanks @dgarcia360  for the response. I have made the changes requested

**Comment by dgarcia360** [comment]
Have done some final formatting editions, the scenarios were fine. After the continuous integration linter passes, I'll merge the PR 👍

**Pull request merged**